azure-mgmt-batch/src/main/java/com/microsoft/azure/management/batch/implementation/PoolsImpl.java

```diff
@@ -73,4 +74,13 @@ protected PoolImpl newChildResource(String name) {
     public void addPool(PoolImpl pool) {
         this.addChildResource(pool);
     }
+
+    public Map<String ,Pool> asMap() {

```

Comment:
What's the reason to provide this API?

Reply:
This asMap() will be called internally by batchAccount.pools() when user wants to work on a certain pool in the map. We keep same map structure for application and pool under batch account.